src/fsverity/ioctl.rs

```diff
@@ -63,16 +82,48 @@ pub fn fs_ioc_measure_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<H>
         digest: H::EMPTY,
     };
 
-    unsafe {
+    let r = unsafe {
         ioctl::ioctl(
             fd,
             ioctl::Updater::<FsIocMeasureVerity, FsVerityDigest<H>>::new(&mut digest),
-        )?;
+        )
+    };
+    match r {
+        Ok(()) => {
+            if digest.digest_algorithm != digest_algorithm {
+                return Err(MeasureVerityError::InvalidDigestAlgorithm {
+                    expected: digest.digest_algorithm,
+                    found: digest_algorithm,
+                });
+            }
+            if digest.digest_size != digest_size {
+                return Err(MeasureVerityError::InvalidDigestSize {
+                    expected: digest.digest_size,
+                });
+            }
+            Ok(Some(digest.digest))
+        }
+        // This function returns Ok(None) if there's no verity digest found.
+        Err(Errno::NODATA | Errno::NOTTY) => Ok(None),
+        Err(Errno::OVERFLOW) => Err(MeasureVerityError::InvalidDigestSize {
+            expected: digest.digest_size,
+        }),
+        Err(e) => Err(std::io::Error::from(e).into()),
```

Comment:
Yikes... is this due to some missing linkage between `rustix` `Errno` and `thiserror`?  Is there a reason you used `std::io::Error` in our error struct instead of `Errno`?  Just to hide rustix as an implementation detail?  (That's a good reason...)

Reply:
Yes, if we exposed the rustix one then semver bumps in rustix would become semver bumps for us.

Reply:
That's good to know, thanks.  I'll make sure to think about that going forward...